utils/swift_build_support/swift_build_support/products/swiftpm.py

```diff
@@ -44,7 +45,11 @@ def should_build(self, host_target):
     def run_bootstrap_script(self, action, host_target, additional_params=[]):
         script_path = os.path.join(
             self.source_dir, 'Utilities', 'bootstrap')
-        toolchain_path = self.install_toolchain_path(host_target)
+
+        if self.args.native_swift_tools_path is not None:
+            toolchain_path = os.path.split(self.args.native_swift_tools_path)[0]
+        else:
+            toolchain_path = self.install_toolchain_path(host_target)
```

Comment:
This same incantation shows up above as well. Can it be factored out into some kind of `self.native_toolchain_path(host_target)` operation?

Reply:
Sure.